SoObjects/SOGo/SOGoUserManager.m

```diff
@@ -475,27 +488,22 @@ - (BOOL) checkLogin: (NSString *) _login
               grace: (int *) _grace
            useCache: (BOOL) useCache
 {
-  NSMutableDictionary *currentUser, *failedCount;
+  NSMutableDictionary *currentUser;
+  NSDictionary *failedCount;
   NSString *dictPassword, *username, *jsonUser;
   SOGoSystemDefaults *dd;
   BOOL checkOK;
- 
-  // We check for cached passwords. If the entry is cached, we 
-  // check this immediately. If not, we'll go directly at the
-  // authentication source and try to validate there, then cache it.
-  if (*_domain != nil)
+
+  if (*_domain && [_login rangeOfString: @"@"].location == NSNotFound)
     username = [NSString stringWithFormat: @"%@@%@", _login, *_domain];
   else
     username = _login;
 
-  failedCount = [[SOGoCache sharedCache] failedCountForLogin: username];
-  dd = [SOGoSystemDefaults sharedSystemDefaults];
-
-  //
   // We check the fail count per user in memcache (per server). If the
   // fail count reaches X in Y minutes, we deny immediately the
   // authentications for Z minutes
-  //
+  failedCount = [[SOGoCache sharedCache] failedCountForLogin: username];
+  dd = [SOGoSystemDefaults sharedSystemDefaults];
```

Comment:
this will change once we merge inverse/master so not really important, though I agree with your suggestion